Dockerfiles/midstream/run_tests.py

```diff
@@ -42,24 +42,43 @@ def run_validate_operator_bundle(self):
                     -e 'work_dir={work_dir}'".format(operator_dir=self.operator_dir,
                                                     operator_work_dir=self.operator_work_dir,
                                                     work_dir=self.work_dir)
-        result = subprocess.run(exec_cmd, shell=True)
+        result = subprocess.run(exec_cmd, shell=True, capture_output=True)
         return result
 
     def test_for_extract_and_validate_bundle_image(self):
+        global exit_code
+        exit_code = 0
         result = self.run_extract_operator_bundle()
-        self.assertEqual(0,
-                         result.returncode,
-                         "extract-operator-bundle.yml failed")
+        if(result.returncode != 0):
+            print("Ansible playbook extract-operator-bundle.yml failed with result code: %s , see the file extract-operator-bundle.errormessage for more info." % result.returncode)
+            with open("extract-operator-bundle.errormessage", "w") as errorMsg:
+                print("Result code: " + str(result.returncode), "Error message: " + str(result.stderr), file=errorMsg)
+            exit_code = 55
+            return exit_code
         result = self.run_validate_operator_bundle()
-        self.assertEqual(0,
-                         result.returncode,
-                         "validate-operator-bundle.yml failed")
+        if(result.returncode != 0):
+            print("Ansible playbook validate-operator-bundle.yml failed with result code: %s , see the file validate-operator-bundle.errormessage for more info." % result.returncode)
+            with open("validate-operator-bundle.errormessage", "w") as errorMsg:
+                print("Result code: " + str(result.returncode), "Error message: " + str(result.stderr), file=errorMsg)
+            exit_code = 60
+            return exit_code
         self.assertTrue(path.exists("/project/output/validation-rc.txt"))
         self.assertTrue(path.exists("/project/output/validation-output.txt"))
         self.assertTrue(path.exists("/project/output/validation-version.txt"))
         with open('/project/output/validation-rc.txt', 'r') as reader:
             rc = reader.read()
-            self.assertEqual(0, int(rc), "validation of bundle image failed")
+            if(int(rc) != 0):
+                print("Image bundle validation failed with result code: %s , see /project/output/validation-output.txt file for more info." % int(rc))
+                exit_code = 70
+                return exit_code
+            #self.assertEqual(0, int(rc), "validation of bundle image failed")
 
 if __name__ == '__main__':
-    unittest.main()
+    tests_result_file = open('tests_result.log','w')
+    unittest.main(testRunner=unittest.TextTestRunner(stream=tests_result_file), exit=False)
+    with open("tests_result.log") as f:
+        if 'F' in f.read():
```

Comment:
Please do not rely on the unittest module here.

We just need simple python script which will execute two ansible playbooks and set expected return codes.
And do something with log files.

Using the unitest module just make things more complicated.

Reply:
This section handles the case in which the unit tests fails on any other occasion than the previous ones, and the **exit=False** needs to be there if we want to send custom return codes.

Reply:
In case we change it to basic unittest.main() method, we would not be ablo to do two things:
send custom return codes, handle errors besides those between 50-99.

Reply:
One more time.
There is not any benefit of using the unitest module for wrapper script. The official documentation says:
```
The unittest unit testing framework was originally inspired by JUnit and has a similar flavor as major unit testing frameworks in other languages.
```
https://docs.python.org/3/library/unittest.html

We do not do any unit testing in the file `run_tests.py` It is just a wrapper script for ansible playbooks because we cannot return different return code from ansible paybook.

@samvarankashyap  did very wrong decision in the initial implementation. But he was sick and we wanted to merge patches to unblock other team.

The usage of the unittest module just makes everything more complicated.
* You have to use global variables. 
* "parse" output from test runner

It will be much simpler without the usage of unittest module

Reply:
@jsztuka PLease focus mainly on requested changes to `unit_tests.py` 

We should firstly focus on tests and than on refactoring ( and it can be done even later together with CVP-1928: Midstream-CVP container yields tests results in a JUnit XML format